pkg/webhooks/admission/hypernodes/validate/admit_hypernode.go

```diff
@@ -84,21 +84,48 @@ func AdmitHyperNode(ar admissionv1.AdmissionReview) *admissionv1.AdmissionRespon
 }
 
 // validateHyperNodeMemberSelector is to validate hypernode member selector.
-func validateHyperNodeMemberSelector(selector hypernodev1alpha1.MemberSelector, fldPath *field.Path) field.ErrorList {
+func validateHyperNodeMemberSelector(selector hypernodev1alpha1.MemberSelector, fldPath *field.Path, memberType hypernodev1alpha1.MemberType) field.ErrorList {
 	errs := field.ErrorList{}
 
-	if selector.RegexMatch == nil && selector.ExactMatch == nil {
-		err := field.Invalid(fldPath, selector,
-			"member selector must have one of regexMatch or exactMatch")
+	if memberType != hypernodev1alpha1.MemberTypeHyperNode && memberType != hypernodev1alpha1.MemberTypeNode {
+		err := field.Invalid(fldPath, memberType, "hyperNode member type must be one of HyperNode or Node")
 		errs = append(errs, err)
 		return errs
 	}
-	if selector.RegexMatch != nil && selector.ExactMatch != nil {
-		err := field.Invalid(fldPath, selector,
-			"member selector cannot have both regexMatch and exactMatch")
-		errs = append(errs, err)
-		return errs
+	if memberType == hypernodev1alpha1.MemberTypeHyperNode {
+		if selector.RegexMatch == nil && selector.ExactMatch == nil {
+			err := field.Invalid(fldPath, selector,
+				"member selector must have one of regexMatch or exactMatch when member type is HyperNode")
+			errs = append(errs, err)
+			return errs
+		}
+	} else {
+		if selector.RegexMatch == nil && selector.ExactMatch == nil && selector.LabelMatch == nil {
+			err := field.Invalid(fldPath, selector,
+				"member selector must have one of regexMatch or exactMatch or labelMatch when member type is Node")
+			errs = append(errs, err)
+			return errs
+		}
+	}
+
+	if memberType == hypernodev1alpha1.MemberTypeHyperNode {
+		if selector.LabelMatch != nil || (selector.RegexMatch != nil && selector.ExactMatch != nil) {
+			err := field.Invalid(fldPath, selector,
+				"member selector cannot have labelMatch or both regexMatch and exactMatch when member type is HyperNode")
+			errs = append(errs, err)
+			return errs
+		}
+	} else {
+		if (selector.LabelMatch != nil && (selector.RegexMatch != nil || selector.ExactMatch != nil)) ||
```

Comment:
This is a bit too complext to understand, making it difficult for others to read. Esstentially, we want to verify that there can not be two or more matches. Perhaps we can use a counting method to do the same thing.